### DISCUSSION TITLE
建议PAGImageView setPath 能在 CacheDir 中，自定义缓存目录

### DISCUSSION THREAD

**Discussion by xyggs** [Tencent/libpag#1894] [Ideas] [answered]
我们在部分场景中，会展示大量的PAGImageView, 全部缓存下来有将近1个G的大小，我们希望在用户离开或者在特定的时机中，清理掉这部分缓存，目前PAGDiskCache中只提供了 RemoveAll 方法。会一起清理掉我们常驻场景中的 pagImageView 缓存

**Comment by kevingpqi123** [comment, marked as answer]
如果是这样的场景，全部缓存将近 1 个 GB，其实不太适合使用 PAGImageView，我们推出 PAGImageView 的目的是希望持久化存储渲染缓存结果，下次使用的时候可以直接加载，如果用过之后就清除了还不如直接使用实时渲染方案 PAGView

**Reply by xyggs** [reply]
我们场景下是一个 3401 * 2436 的背景中，里面分布了8 - 11个 动画的view。大小平均在 200 * 200 dp 如果用pagView的话，一直是卡顿的，发热也很大。用了pagImageView 的话。缓存完毕之后就流畅了，但是 缓存又太大了，产品又觉得这样用户会有不小的意见，所以才用过之后就清除。

**Comment by xyggs** [comment]
后续我们还会有相同的场景，比如 一个大房间里会容纳 多个动画的人物view，可以自由的通过图层替换来达到换装效果。如果在这种场景下用pagView， 因该会很卡很烫。

**Reply by kevingpqi123** [reply]
那核心问题是优化这个场景下的性能，对于 PAGImageView 而言，虽然缓存结束后后续加载速度和性能提升很大，但解决不了首次加载的问题，首次加载的时候本质上还需要去渲染
推荐几个优化思路：
1、合并渲染：如果一个页面中含有多个 PAGView，尽量合并到 1 个 PAGView 中，通过 PAG 的组合模式将多个 pag 文件组合在一起，可以减少每一个 PAGView 的 OpenGL 渲染环境的基础开销
2、优化素材：尽可能的使用纯矢量的素材

**Reply by xyggs** [reply]
1.如果合并成一个pagView的话，我们这个pagView 尺寸可能会非常非常大
2.我们每个pagView 都有自己的点击相应，组合后，还能不同的pag响应不同的点击事件吗，或者说 单独控制哪些动画播放 哪些不播放

**Reply by kevingpqi123** [reply]
PAGView 本来就支持全屏播放，每个 PAGView 的创建都会涉及 OpenGL 基础环境的创建，这里的内存开销和CPU占用不小，所以要尽可能的减少一个页面中的 PAGView 的数量
关于第二个问题，了解下 PAGView getLayersUnderPoint 接口，就是为了解决某一点含有哪些图层的问题，关于单独控制播放，更不是问题，通过setStartTime可以设置相当于主时间轴的起始播放时间，通过PAGPlayer 的 setVisible 可以控制图层或单个 pag 文件是否显示

**Comment by peter100u** [comment]
缓存的容量不是固定的吗，然后不经常使用的，pag自动清理， 我记得群里说了pag会自动管理缓存的大小
